tools/image_processing/yolo/yolo_training.xml

```diff
@@ -0,0 +1,149 @@
+<tool id="yolo_training" name="Perform YOLO training" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="24.2">
+    <description>with ultralytics</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="creator" />
+    <expand macro="edam" />
+    <expand macro="requirements" />
+    <command detect_errors="aggressive">
+    <![CDATA[
+      export YOLO_CONFIG_DIR='\$HOME/.config/ultralytics' &&
+
+      mkdir -p ./input_images ./input_yolo ./training ./runs ./models output_files && 
+
+      #for $filename in $input_images:
+          ln -s '$filename' './input_images/${filename.element_identifier}' &&
+      #end for
+	
+      #for $yolo in $input_yolo:
+          ln -s '$yolo' './input_yolo/${yolo.element_identifier}' &&
+      #end for
+
+      python '$__tool_directory__/preprocessing.py' -i './input_images' -y './input_yolo' -o ./training -p '$num_train' -m '$input_json' &&
+
+      wget '$training_params.model_url' -O ./models/yolo_model.pt &&
+		    
+      
+      python '$__tool_directory__/yolov8.py' 
+		    --train --yaml_path='./training/yolo.yml' 
+		    --model_path='./models' 
+     		    --model_name='yolo_model'
+		    --run_dir='./runs' 
+		    --image_size='$training_params.image_size'
+		    --epochs='$training_params.epochs'
+		    --scale='$training_params.scale'
+		    --degree='$training_params.degree'
+		    --hsv_v='$training_params.hsv_v'
+		    --hsv_s='$training_params.hsv_s'
+		    --hsv_h='$training_params.hsv_h'
+		    --init_lr='$training_params.init_lr'
+		    --weight_decay='$training_params.weight_decay'
+		    --confidence='$training_params.confidence'
+		    --iou='$training_params.iou'
+		    --max_det='$training_params.max_det'  
+
+      &&
+
+      cp ./runs/train*/*/best.pt output_files/best.pt &&
+      cp ./runs/train*/*/last.pt output_files/last.pt &&
+      cp ./runs/train*/results.csv output_files/results_metrics.csv &&
+      cp ./runs/train*/results.png output_files/results_plot.png 
+    ]]>
+    </command>
+    <configfiles>
+	 <inputs name="input_json"/>
+    </configfiles>
+    <inputs>
+        <param name="input_images" type="data" format="jpg,png,tiff" multiple="true"  label="Input images"/>
```

Comment:
> > **2)** I think you should also add to the help section, that only 2-D images are supported. What about multi-channel images, like RGB images? If not supported, please explicitly mention this in the help section.
> 
>  […]
> * 2,3.  I couldn't find any docs explicitly says this, but it seems all yolo models are trained using jpeg.

The tool accepts `png` and `tiff` besides of `jpg`, but the tests only cover the `jpg` case. Can you maybe add a (small) test for `png` and `tiff`, just so that we can be sure that the wrapper does handle conversion properly?

Or, alternatively:
```suggestion
        <param name="input_images" type="data" format="jpg" multiple="true"  label="Input images"/>
```